From 4952b23f64e73f169d301d61e6c93f368b03edb6 Mon Sep 17 00:00:00 2001
From: Wallace Wang <r59996@freescale.com>
Date: Wed, 5 Nov 2008 13:39:19 +0800
Subject: [PATCH] ENGR00097749-1 MX51: Fix spdif gpio free error

Request the gpio with ALT2, but free it as GPIO

Signed-off-by: Wallace Wang <r59996@freescale.com
---
 arch/arm/mach-mx51/mx51_3stack_gpio.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-mx51/mx51_3stack_gpio.c b/arch/arm/mach-mx51/mx51_3stack_gpio.c
index 2664f64..70aab1e 100644
--- a/arch/arm/mach-mx51/mx51_3stack_gpio.c
+++ b/arch/arm/mach-mx51/mx51_3stack_gpio.c
@@ -1587,7 +1587,7 @@ EXPORT_SYMBOL(gpio_spdif_active);
 
 void gpio_spdif_inactive(void)
 {
-	mxc_free_iomux(MX51_PIN_GPIO1_7, IOMUX_CONFIG_GPIO);
+	mxc_free_iomux(MX51_PIN_GPIO1_7, IOMUX_CONFIG_ALT2);
 
 }
 
-- 
1.5.4.4

